routing/endpointregistry.go

```diff
@@ -60,23 +73,54 @@ func (e *entry) SetLastSeen(ts time.Time) {
 	e.lastSeen.Store(ts)
 }
 
-func newEntry() *entry {
+func (e *entry) IncRequests(o IncRequestsOptions) {
+	curSlot := e.curSlot.Load()
+	e.totalRequests[curSlot].Add(1)
+	if o.FailedRoundTrip {
+		e.totalFailedRoundTrips[curSlot].Add(1)
+	}
+}
+
+func (e *entry) HealthCheckDropProbability() float64 {
+	totalRequests := e.totalRequests[e.curSlot.Load()].Load()
+	if totalRequests < e.minRequests {
+		return 0.0
+	}
```

Comment:
Why do we need this if calculation of healthCheckDropProbability accounts for this?

Reply:
no we do not, we should just return the value as in line 89

Reply:
We decided on the meeting that I will remove those lines.